File: src/algorithms/check-string.js
/**
 * Given a string s consisting of only the characters 'a' and 'b',
 * returns true if every 'a' appears before every 'b' in the string.
 * Returns false otherwise.
 * 
 * @param {string} s - A string made up of lowercase 'a' and 'b'
 * @returns {boolean} - Returns true if there are no 'b' chars before 'a' chars
 */
function checkString (s) {
  let bAppearance = false;
  for (let i = 0; i < s.length; i++) {
    if (s[i] === 'b') {
      bAppearance = true;
    } else if (bAppearance && s[i] === 'a') {
      return false;
    }
  }
  return true;
}

File: src/algorithms/check-string.py
def check_string(s):
    """Given a string s consisting of only the characters 'a' and 'b', return 
    true if every 'a' appears before every 'b' in the string. Otherwise, return 
    false.

    Args:
        s (str): A string of lowercase "a" and "b" characters.

    Returns:
        _type_: _description_
    """
    b_seen = False
    for char in s:
        if char == "b":
            b_seen = True
        elif b_seen == True:
            return False
    return True
